src/Ultz.SuperInvoke.Loader/LibraryLoader.cs

```diff
@@ -20,9 +20,28 @@ public abstract class LibraryLoader
         /// </summary>
         /// <param name="name">The name of the library to open.</param>
         /// <returns>The operating system handle for the shared library.</returns>
-        public IntPtr LoadNativeLibrary(string name)
+        public IntPtr LaodNativeLibrary(string name)
         {
-            return LoadNativeLibrary(name, PathResolver.Default);
+            var success = TryLoadNativeLibrary(name, out IntPtr result);
+
+            if (!success)
+                throw new FileNotFoundException("Could not find or load the native library: " + name);
+
+            return result;
+        }
+
+        /// <summary>
+        ///     Loads a native library by name and returns an operating system handle to it.
+        /// </summary>
+        /// <param name="name">The name of the library to open.</param>
+        /// <param name="result">A pointer to the loaded library.</param>
+        /// <returns>The operating system handle for the shared library.</returns>
+        public bool TryLoadNativeLibrary(string name, out IntPtr result)
+        {
+            var success  = TryLoadNativeLibrary(new[] { name }, PathResolver.Default, out IntPtr libPtr);
```

Comment:
You likely don’t need to declare another variable for this, just pass result directly as out.

Reply:
fixed